lib/DxilValidation/DxilValidation.cpp

```diff
@@ -1229,6 +1229,32 @@ static void ValidateImmOperandsForOuterProdAcc(CallInst *CI,
         {"MatrixLayout"});
     return;
   }
+  ConstantInt *ML = cast<ConstantInt>(MatrixLayout);
+  uint64_t MLValue = ML->getLimitedValue();
+  if (MLValue !=
+      static_cast<unsigned>(DXIL::LinalgMatrixLayout::OuterProductOptimal))
+    ValCtx.EmitInstrFormatError(
+        CI,
+        ValidationRule::
+            InstrLinalgInvalidMatrixLayoutValueForOuterProductAccumulate,
+        {GetMatrixLayoutStr(MLValue),
+         GetMatrixLayoutStr(static_cast<unsigned>(
+             DXIL::LinalgMatrixLayout::OuterProductOptimal))});
+
+  llvm::Value *MatrixStride =
+      CI->getOperand(DXIL::OperandIndex::kOuterProdAccMatrixStride);
+  if (!llvm::isa<llvm::Constant>(MatrixStride)) {
+    ValCtx.EmitInstrError(
+        CI, ValidationRule::InstrLinalgMatrixStrideZeroForOptimalLayouts);
```

Comment:
Shouldn't this be a separate error? One error that triggers when the value isn't 0, and one when the value isn't constant?
Using the same diagnostic could work, but perhaps an improvement to the message would be to specify that the 0 has to be a constant.

Reply:
So matrix stride is in general not expected to be a constant value, but for optimal formats it has to be a constant zero value, hence I structured the error this way. I can change the error to specify that the value should be cosntant zero for optimal layouts if you prefer. Also for this and the comment below do you want it in a separate PR once this is merged?